agent/src/main/java/org/wso2/testgrid/agent/AgentApplication.java

```diff
@@ -0,0 +1,132 @@
+import javax.websocket.CloseReason;
+
+/**
+ * Agent bootstrap class.
+ *
+ * @since 1.0.0
+ */
+public class AgentApplication implements OperationResponseListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(AgentApplication.class);
+    private static final String AGENT_PROP_FILE = File.separator + "opt" + File.separator + "testgrid" +
+            File.separator + "agent-config.properties";
+
+    private ClientEndpoint clientEndPoint;
+
+    public static void main(String args[]) {
+        new AgentApplication().startService();
+        while (true) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                break;
+            }
+        }
+    }
+
+    private void startService() {
+        String wsEndpoint;
+        String agentId;
+
+        Properties prop = new Properties();
+        InputStream input = null;
+        try {
+            if (new File(AGENT_PROP_FILE).exists()) {
+                input = new FileInputStream(AGENT_PROP_FILE);
+                prop.load(input);
+                wsEndpoint = prop.getProperty("wsEndpoint");
+                //aws:us-west-1:6718f976-7150-316d-9ec6-db29b7a2675f:i-0658922f45d25856f
+                agentId = prop.getProperty("provider") + ":" + prop.getProperty("region") + ":" +
+                        prop.getProperty("testPlanId") + ":" + prop.getProperty("instanceId");
+            } else {
+                logger.warn("Agent configurations not found.");
+                return;
+            }
+        } catch (IOException ex) {
+            logger.error("Error occurred when reading prop file: " + ex.getMessage(), ex);
+            return;
+        } finally {
+            if (input != null) {
+                try {
+                    input.close();
+                } catch (IOException e) {
+                    logger.error("Error occurred when closing prop file stream: " + e.getMessage(), e);
+                }
+            }
+        }
+
+        try {
+            logger.info("Agent ID: " + agentId);
+            // open web socket
+            clientEndPoint = new ClientEndpoint(new URI(wsEndpoint + agentId));
+
+            clientEndPoint.addClientHandler(new ClientEndpoint.ClientHandler() {
+                @Override
+                public void handleMessage(String message) {
+                    logger.info("Operation received: " + message);
+                    OperationRequest operationRequest = new Gson().fromJson(message, OperationRequest.class);
+                    OperationExecutor operationExecutor = new OperationExecutor(AgentApplication.this);
+                    operationExecutor.executeOperation(operationRequest);
+                }
+
+                @Override
+                public void handleClose(CloseReason reason) {
+                    logger.warn("Client disconnected. Retrying to connect.");
```

Comment:
Shall we print the CloseReason?

Reply:
It is already printed with onClose call back in ClientEndpoint.